### PR TITLE
Add details to vmsize list and azure location endpoint

### LINKED ISSUE #10713: Add GET endpoint for Azure locations
### Description of the feature you would like to add / User story

As a user
I would like to Add GET endpoint for Azure locations
in order to utilize it in AKS cluster creation

### Solution details

- Requirement 1
- Requirement 2
- Requirement 3

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What this PR does / why we need it**:
- add GET endpoint for Azure Recommended locations
- add Instancetype details like memory, vcpu to GET vmsizes instance for EKS and AKS
- refactor ValidateCredentials error message to be more clear
- move a few methods and constants to other files
- rename isImported label to `is-imported`

**Which issue(s) this PR fixes**:

Fixes #10713

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

**Comment by embik** [comment]
@imharshita FYI, the documentation field is not intended to be used that way. Did you mean to put release notes?

File: go.mod [hunk 1]
```diff
@@ -20,7 +20,7 @@ require (
 	github.com/aws/aws-sdk-go v1.44.57
 	github.com/aws/aws-sdk-go-v2/service/eks v1.21.4
 	github.com/cert-manager/cert-manager v1.9.1
-	github.com/cilium/cilium v1.12.0 // need 1.12.x because 1.11.x is incompatible with k8s 1.24
+	github.com/cilium/cilium v1.11.0-rc0 // need 1.12.x because 1.11.x is incompatible with k8s 1.24
```
Comment (embik) [inline go.mod]: Please don't downgrade cilium.

File: pkg/handler/v2/routes_v2.go [hunk 2]
```diff
@@ -1161,16 +1165,16 @@ func (r Routing) createCluster() http.Handler {
...
-//       default: errorResponse
-//       200: ClusterList
-//       401: empty
-//       403: empty
+//	Responses:
+//	  default: errorResponse
+//	  200: ClusterList
+//	  401: empty
+//	  403: empty
 func (r Routing) listClusters() http.Handler {
```
Comment (embik) [inline pkg/handler/v2/routes_v2.go]: This whole file is getting formatted, can you please undo that to reduce the diff of this PR?

**Comment by imharshita** [comment]
> @imharshita FYI, the documentation field is not intended to be used that way. Did you mean to put release notes?

okay, can u plz share why is it added?

File: pkg/handler/v2/routes_v2.go (hunk 2 above)
Comment (imharshita) [reply pkg/handler/v2/routes_v2.go]: cool, it was auto done by vscode and even goland, I couldn't figure out why.
But, now I removed then using text editor, only kept the change of `//swagger => // swagger` to keep it consistent

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**